src/stores/pacbioTubes.js

```diff
@@ -0,0 +1,50 @@
+import { defineStore } from 'pinia'
+import handlePromise from '@/api/PromiseHelper'
+import store from '@/store'
+
+export const usePacbioTubesStore = defineStore('pacbioTubes', {
+  state: () => {
+    return {
+      tractionTubes: [],
+    }
+  },
+  getters: {
+    //Pinia_migration_todo: This is migrated from the VueX store, but it needs to be changed to  Pinia store once the VueX store is removed
+    tubeRequest: () => store.state.api.traction.pacbio.tubes,
+    //Pinia_migration_todo: This is migrated from the VueX store, but it needs to be changed to  Pinia store once the VueX store is removed
+    requestsRequest: () => store.state.traction.pacbio.requests,
+  },
+
+  actions: {
+    async isLibraryBarcodeValid(barcode) {
+      if (!barcode) {
+        return false
+      }
+      const libraryTube = await this.getTubeForBarcode(barcode)
+      return this.validateLibraryTube(libraryTube)
+    },
+    async getTubeForBarcode(barcode) {
+      const request = this.tubeRequest
+      const promise = request.get({ filter: { barcode: barcode }, include: 'pools' })
+      const response = await handlePromise(promise)
```

Comment:
We are trying to move away from `handlePromise`. We now use `handleResponse` from src/api/ResponseHelper. Do you think you could change this function to use handleResponse instead?